[TEST] Migrated REST tests extending JsonLogsIntegTestCase

jozala opened the pull request:
REST tests extending JsonLogsIntegTestCase migrated to the new REST testing framework, using 'elasticsearch.internal-java-rest-test' Gradle plugin

jozala committed:
[TEST] Migrated REST tests extending JsonLogsIntegTestCase

REST tests extending JsonLogsIntegTestCase migrated to the new REST
testing framework, using 'elasticsearch.internal-java-rest-test' Gradle
plugin

jozala commented on qa/unconfigured-node-name/src/javaRestTest/java/org/elasticsearch/unconfigured_node_name/JsonLogsFormatAndParseIT.java:
```diff
@@ -30,6 +31,30 @@ public class JsonLogsFormatAndParseIT extends JsonLogsIntegTestCase {
+        .setting("xpack.security.enabled", "false")
+        .withNode(localNodeSpecBuilder -> localNodeSpecBuilder.name(OS_NAME.startsWith("Windows") ? COMPUTERNAME : HOSTNAME))
+        // TODO @jozala -
+        .settingsModifier(
```
I've added this settings modifier to be able to remove `node.name` from configuration, but keep the rest of the Elasticsearch node setup working. I am wondering if there is already a better way to achieve this.

jozala commented on distribution/archives/integ-test-zip/src/javaRestTest/java/org/elasticsearch/test/rest/JsonLogsFormatAndParseIT.java:
```diff
@@ -10,36 +10,45 @@
 
     @Override
     protected BufferedReader openReader(Path logFile) {
-        assumeFalse(
```
I haven't recreated this behaviour. Do we run these tests against external clusters? Do we need this?

jozala commented on distribution/archives/integ-test-zip/src/javaRestTest/java/org/elasticsearch/test/rest/JsonLogsFormatAndParseIT.java:
```diff
@@ -10,36 +10,45 @@
 
     @Override
     protected BufferedReader openReader(Path logFile) {
-        assumeFalse(
-            "Skipping test because it is being run against an external cluster.",
-            logFile.getFileName().toString().equals("--external--")
-        );
-        return AccessController.doPrivileged((PrivilegedAction<BufferedReader>) () -> {
-            try {
-                return Files.newBufferedReader(logFile, StandardCharsets.UTF_8);
-            } catch (IOException e) {
-                throw new RuntimeException(e);
-            }
-        });
```
I haven't recreated this behaviour. Do we run these tests against external clusters? Do we need this?

jozala replied:
Discussed with @breskeby and believe we do not run these tests against external clusters.
In theory when running the tests against external cluster some tests would be ignored using this code.
I have removed this for now and I will get back to it if see some tests are failing.

jozala commented on qa/unconfigured-node-name/src/javaRestTest/java/org/elasticsearch/unconfigured_node_name/JsonLogsFormatAndParseIT.java:
```diff
@@ -30,6 +31,30 @@ public class JsonLogsFormatAndParseIT extends JsonLogsIntegTestCase {
+        .setting("xpack.security.enabled", "false")
+        .withNode(localNodeSpecBuilder -> localNodeSpecBuilder.name(OS_NAME.startsWith("Windows") ? COMPUTERNAME : HOSTNAME))
+        // TODO @jozala -
+        .settingsModifier(
+            settings -> settings.entrySet()
+                .stream()
+                .filter(it -> it.getKey().equals("node.name") == false)
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))
+        )
```
I've added this settings modifier in configuration to be able to remove `node.name` from configuration, but keep the rest of the Elasticsearch node setup working. I am wondering if there is already a better way to achieve this.

jozala committed:
Set settings modifier to be an identity function by default

jozala committed:
Change approach to testing unnamed node

Allow unsetting the node name on the cluster spec instead of allowing to
 override any setting in the node configuration.

jozala committed:
Merge branch 'main' into json-logs-tests-migration

jozala committed:
Merge branch 'main' into json-logs-tests-migration

jozala committed:
Dependency to `assemble` tasks re-added

Fix failing serverless build with error:
```
Incompatible because this component declares attribute 'artifactType'
with value 'jar' and the consumer needed attribute 'artifactType' with
value 'directory'
```

jozala commented on test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/DefaultSettingsProvider.java:
```diff
@@ -70,17 +76,20 @@ public Map<String, String> get(LocalNodeSpec nodeSpec) {
-                "Cannot start cluster '" + nodeSpec.getCluster().getName() + "' as it configured with no master-eligible nodes."
+            LOGGER.warn(
+                "Cluster configured with no master-eligible nodes. It may cause split-brain if more than one node is in the cluster."
             );
         }
 
-        settings.put("cluster.initial_master_nodes", "[" + masterEligibleNodes + "]");
-        settings.put("discovery.seed_providers", "file");
-        settings.put("discovery.seed_hosts", "[]");
+        if (masterEligibleNodes.isEmpty() == false) {
+            settings.put("cluster.initial_master_nodes", "[" + masterEligibleNodes + "]");
+            settings.put("discovery.seed_providers", "file");
+            settings.put("discovery.seed_hosts", "[]");
+        }
```
@mark-vieira @breskeby 
This is something I would definitely like your opinion on. I am now allowing creating cluster without setting the `cluster.initial_master_nodes` where it was explicitly disallowed previously. I believe this is OK when there is a single node, but it possible can create some problems in more complex tests. 
I need that to be able to start the node without the explicit name in the `JsonLogsFormatAndParseIT` test.

mark-vieira replied:
Well, in the case where the node doesn't have a name, `masterEligibleNodes` here would be an empty string, no? Would setting `initial_master_nodes` to `[]` be equivalent to not setting it at all or is the behavior different?

mark-vieira replied:
Alternatively, could we make that above error condition more specific. That is, only throw an exception when there are no master eligible nodes _and_ the cluster contains multiple nodes?

jozala committed:
Packages with migrated tests removed from exclusion list

jozala committed:
Remove unused StandaloneRestIntegTestTask configuration

jozala committed:
Do not expose runtimeElements and apiElements in build-integ-zip

This is to avoid using these configurations in variant resolution where
artifactType of "directory" is needed

jozala committed:
More explicit handling of single-cluster config

Checking if cluster is explicitly configured as single-node to avoid an
incorrect configuration which may cause split brain.

jozala committed:
Merge branch 'main' into json-logs-tests-migration

jozala committed:
Node suffix configuration restored

I've removed node suffix configuration previously as it seemed unused,
but it is required for some tests.

jozala committed:
Revert "Do not expose runtimeElements and apiElements in build-integ-zip"

This reverts commit dd4c4bc4339d04f0af2e8cbb377c6efb18384954.

jozala committed:
Code formatted properly

jozala committed:
Set consumption and resolution properties for configurations

Updated the `DistributionDownloadPlugin` to set configurations as
non-consumable and the `InternalDistributionArchiveSetupPlugin` to mark
certain configurations as consumable. This ensures that configurations
are correctly utilized during build setup and execution phases.

jozala committed:
Merge branch 'main' into json-logs-tests-migration

jozala committed:
Apply ElasticsearchJavaBasePlugin to RestTestBasePlugin

Replaced `ElasticsearchJavaPlugin` with `ElasticsearchJavaBasePlugin` in
 `RestTestBasePlugin` for better granularity.
Additionally, removed unused `compileOnly` and `testImplementation`
dependencies from several build.gradle files using `RestTestBasePlugin`
where these configurations are no longer available.

jozala committed:
Merge branch 'elastic:main' into json-logs-tests-migration

breskeby commented on x-pack/plugin/ent-search/qa/full-cluster-restart/build.gradle:
```diff
@@ -13,7 +13,6 @@ apply plugin: 'elasticsearch.bwc-test'
 
 
 dependencies {
-  testImplementation project(path: xpackModule('ent-search'))
```
what's this about? not needed?

elasticsearchmachine commented:
Pinging @elastic/es-delivery (Team:Delivery)

breskeby reviewed (approved):
I think this looks good. I left one remark for double checking for cleanup but overall this is good to merge.

With having the struggle with the variant selection, I think the RestTestBasePlugin looks also better now being more idiomatic to what a base plugin in gradle terms would do, meaning just provide infrastructure to the build but don't be opinionated or add certain assumptions. 

Also having less logic in our build scripts makes things way easier to maintain for the team. 👍
